### DISCUSSION TITLE
Best way to use Collections (and how to add some word lists)

### DISCUSSION THREAD

**Discussion by manzomium** [adieyal/sd-dynamic-prompts#93] [Q&A] [answered]
Hello

My first post ever on GItHub.
I'm impressed by the potential of your work, and while I started creating my own lists of wildcards, I've just discovered that you provide us with a huge set. However, I'm not sure how to use them at their best. For now, I've copied all the *.txt files from Gdrive into the extensions/wildcard folder. There are now almost 450 items, and it's overwhelming. Some questions:
- Could I have just copied the folders are they were in the Collections folder, then the extension would have searched for them?
- On the other hand, I did this because I wanted to have an overview of all the possible lists I could use. Is there a list of lists somewhere?
- It seems you don't have nationalities lists. I have one for the world, one for Europe, one for Africa. I can share them if you want, let me know how :-)
Thanks!

**Comment by adieyal** [comment, marked as answer]
I think the collections library is underutilised, I have a lot of fun playing around with it. It isn't close to being complete, and likely never will be, but it's quite useful. Your nationalities list would be appreciated.

Regarding your questions:

1. Yes, you can simply copy the entire jumbo or parrot folder from the collections directory into wildcards. The reason it isn't there by default is to avoid overwriting any of your customisations.
2. Have a look at the Wildcards Manager tab, there you'll see a list of lists (a directory tree actually). You can also edit wildcard files. It's a new feature so very rudimentary, I'll be adding additional functionality for creating new files, renaming, deleting, etc in the coming week or two.

You may be interested in this thread: [https://github.com/adieyal/sd-dynamic-prompts/issues/28](https://github.com/adieyal/sd-dynamic-prompts/issues/28). As @Yaiol mentions, many of the artists might not even be recognised by SD. The rationale behind trying to be exhaustive is that future models may indeed recognise a wider range of artists. In any case, the collections are there for you to explore, and tailor for your own purposes.

![Screenshot from 2022-12-04 23-13-47](https://user-images.githubusercontent.com/420786/205516003-5bcc00ac-a120-4d48-9c2f-9806ee2d4f94.png)

**Reply by manzomium** [reply]
Thanks a lot for replying and for the additional info about 'verified' artists. I'll see what I get.
I attach here the nationalities lists I've created so far. I've only made three subdivisions, feel free to do more - or maybe I'll do when I have some more free time.
Thinking about it, divisions could also be done by ethnicity, as some far-east Europeans look like Central Asians, and Northern Africans look similar to East Asian/Arabians. Europe itself could be divided up into Northern Central, Latin/Mediterranean, Slavic/Eastern, Far-Eastern...)
[all-nationalities.txt](https://github.com/adieyal/sd-dynamic-prompts/files/10158147/all-nationalities.txt)
[europe-nationalities.txt](https://github.com/adieyal/sd-dynamic-prompts/files/10158148/europe-nationalities.txt)
[african-nationalities.txt](https://github.com/adieyal/sd-dynamic-prompts/files/10158149/african-nationalities.txt)

Again, let's see how much SD is trained for in terms of ethnicity, nationalities - but so far I got reliable results (although I'm not sure some melting-pot nationalities like English - with massive immigration dating back in time would output darker-skinned people in a way that's representative of the reality on the ground. More to be experimented with...
